include/arch/x86/linker.ld

```diff
@@ -329,6 +341,18 @@ SECTIONS
 #endif /* LINKER_PASS2 */
 #endif /* CONFIG_GDT_DYNAMIC */
 
+#ifdef CONFIG_X86_KPTI
+	z_trampoline_stack_start = .;
+	MMU_PAGE_ALIGN
+	z_trampoline_stack_end = .;
+	z_shared_kernel_page_end = .;
+
+	ASSERT(z_trampoline_stack_end - z_trampoline_stack_start >= 40,
+		"trampoline stack too small");
+	ASSERT(z_shared_kernel_page_end - z_shared_kernel_page_start == 4096,
+	       "shared kernel area is not one memory page");
```

Comment:
Not a problem in practice, but in theory a fully populated IDT and all that other stuff would (I think) be able to overflow.  Might need to move to a multi-page area at some point.

Reply:
I think I'm OK, a full size IDT is 2KB, the remaining data structures much smaller. It should all fit in a page. We'll of course know if it doesn't.